packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -815,7 +815,9 @@ class CupertinoTextField extends StatefulWidget {
     BuildContext context,
     EditableTextState editableTextState,
   ) {
-    if (defaultTargetPlatform == TargetPlatform.iOS && SystemContextMenu.isSupported(context)) {
+    if (defaultTargetPlatform == TargetPlatform.iOS &&
+        SystemContextMenu.isSupported(context) &&
+        !editableTextState.widget.readOnly) {
```

Comment:
It might be helpful to have a comment here on why the system context menu is not supported with `readOnly` set to true.

Reply:
Added on SystemContextMenu.isSupportedByField.